MQ2FeedMe.cpp

```diff
@@ -126,214 +312,370 @@ void ListTypes(const std::list<std::string>& list)
+{
+	if (!strlen(szLine))
+	{
+		if (GoodToConsume())
+		{
+			Consume(ItemClass_Drink, Thirst);
+		}
+		return;
 	}
-	else if (!_stricmp(Arg, "announceConsume")) {
-		if (bAnnConsume) {
-			bAnnConsume = false;
-			WriteChatf("\ay%s\aw::Consumption Notification Off", PLUGIN_NAME);
+
+	char Arg[MAX_STRING] = { 0 };
+	GetArg(Arg, szLine, 1);
+
+	if (ci_equals(Arg, "list"))
+	{
+		WriteChatf("\ay%s\aw:: Listing Drink:", PLUGIN_NAME);
+		ListTypes(Thirst);
+		if (bAnnLevels)
+		{
+			sprintf_s(Buffer, "%d", iDrinkAt);
+			WriteChatf("\ay%s\aw:: AutoDrink (\ag%s\ax).", PLUGIN_NAME, (iDrinkAt) ? Buffer : "\aroff");
+			WriteChatf("\ay%s\aw:: Current Thirst (\ag%d\ax)", PLUGIN_NAME, GetPcProfile()->thirstlevel);
+		}
+	}
+	else if (ci_equals(Arg, "warn"))
+	{
+		if (bDrinkWarn)
+		{
+			bDrinkWarn = false;
+			WriteChatf("\ay%s\aw:: Drink Warning Off", PLUGIN_NAME);
 		}
 		else {
-			bAnnConsume = true;
-			WriteChatf("\ay%s\aw::Consumption Notification On", PLUGIN_NAME);
+			bDrinkWarn = true;
+			WriteChatf("\ay%s\aw:: Drink Warning On", PLUGIN_NAME);
+		}
+	}
+	else if (ci_equals(Arg, "add"))
+	{
+		AddDrinkItem();
+	}
+	else if (IsNumber(Arg))
+	{
+		iDrinkAt = GetIntFromString(Arg, 5000);
+		if (iDrinkAt < 0)
+		{
+			iDrinkAt = 0;
+		}
+		else if (iDrinkAt > 5000)
+		{
+			iDrinkAt = 5000;
+		}
+		sprintf_s(Buffer, "%d", iDrinkAt);
+		WritePrivateProfileString(GetCharInfo()->Name, "AutoDrink", Buffer, INIFileName);
+		WriteChatf("\ay%s\aw:: AutoDrink (\ag%s\ax).", PLUGIN_NAME, iDrinkAt ? Buffer :"\aroff");
+
+		if (bAnnLevels)
+		{
+			WriteChatf("\ay%s\aw:: Current Thirst (\ag%d\ax) Hunger (\ag%d\ax)", PLUGIN_NAME, GetPcProfile()->thirstlevel, GetPcProfile()->hungerlevel);
 		}
 	}
+	else
+	{
+		GenericCommand(szLine);
+	}
 }
 
 PLUGIN_API void OnPulse()
 {
 	static int Pulses = 0;
-	if (++Pulses < 50) return;
+	if (++Pulses < 50)
+	{
+		return;
+	}
+
 	Pulses = 0;
-	if (!GoodToFeed()) return;
-	if (DrnkAt && (LONG)GetPcProfile()->thirstlevel < DrnkAt) Consume(ItemClass_Drink,Thirst);
-	if (FeedAt && (LONG)GetPcProfile()->hungerlevel < FeedAt) Consume(ItemClass_Food,Hunger);
+
+	if (!GoodToConsume())
+	{
+		return;
+	}
+
+	if (iDrinkAt && GetPcProfile()->thirstlevel < iDrinkAt)
+	{
+		Consume(ItemClass_Drink, Thirst);
+	}
+
+	if (iFeedAt && GetPcProfile()->hungerlevel < iFeedAt)
+	{
+		Consume(ItemClass_Food, Hunger);
+	}
 }
 
-PLUGIN_API DWORD OnIncomingChat(PCHAR Line, DWORD Color)
+PLUGIN_API DWORD OnIncomingChat(const char* Line, const unsigned int Color)
 {
-	if (!IAmCamping && strstr(Line, "It will take you about 30 seconds to prepare your camp.")) {
-		IAmCamping = true;
+	if (!bIAmCamping && ci_find_substr(Line, "It will take you about 30 seconds to prepare your camp.") != -1)
+	{
+		bIAmCamping = true;
 	}
-	else if (IAmCamping && strstr(Line, "You abandon your preparations to camp.")) {
-		IAmCamping = false;
+	else if (bIAmCamping && ci_find_substr(Line, "You abandon your preparations to camp.") != -1)
```

Comment:
Also...why is THIS case insensitive? (But also you have an if and an else if that are specifically checking the same variable, so you probably have something you can do here to tighten this up instead of checking it in succession).